modules/engage-paella-player-7/src/js/PaellaOpencast.js

```diff
@@ -142,6 +146,10 @@ const initParams = {
       if (!me.roles.includes('ROLE_USER')) {
         player.log.info('Video not found and user is not authenticated. Try to log in.');
         location.href = getUrlFromOpencastPaella('auth.html?redirect=' + encodeURIComponent(window.location.href));
+        // We need to interrupt the player loading to redirect the user to the auth page.
+        await sleep(100);
+        // This Error should not happen, as the user is redirected to the auth page.
+        throw Error('Video not found and user is not authenticated. Try to log in.');
       }
```

Comment:
Hi @miesgre, our site has tried this same approach in the same part of the code. We still see the red error circle flashing prior to redirecting the page. It is as if the browser requires the current running Javascript function to complete prior to addressing the `location.href` early on. We also wrap promises around everything to try break the execution chain, but still get the red error flash.

Is it possible to throw an exception, like the `throw Error`, to break the execution chain after the `location.href` request, but when caught, the red error UI artifact is not shown?

Another idea is to create a UI artifact that obscures the red error UI artifact, so that when the Error is thrown and breaks the execution chain, allowing the browser to execute the `location.href`, no red Error UI artifact shows prior to the `location.href` being executed.

Reply:
In summary, the `location.href()` request is not executed in-line, at least in browsers like Chrome. It is held in queue until the end of the Javascript execution chain. Until after the `throw Error()` at the end of this code sequence.

Reply:
[EDIT] Our site takes longer to for the `location.href` to resolve because we do an extra redirect. Opencast adopters that use the regular Opencast login might not have the same lag time response from `location.href` as we do.

Adding this `sleep(100)` might be sufficient to prevent the Error artifact from displaying for the default login page redirect.

Reply:
@karendolan , please test ithis exact code in your installation. I think it should word. The time in the sleep does not matter. The important is to break the execution chain because the location.href is not executed immediately..

The throw after the sleep is **never** executed!. I put it only in case one day browsers change this behavior.

Even you can change the sleep function for a non resolve promise function if you want.

```js
function nonResolvingPromise() {
  return new Promise(() => {});
}

...
await nonResolvingPromise();
```

Reply:
@miesgre What is different about our site is that the` href.location` does a redirect chain. It ` returns an HTTP 301, then HTTP 302, then HTTP 200 on the redirected page. This is different than the default Opencast login page which returns a HTTP 200 immediately. Our sites issue could be related to the back and forth of the HTTP 301.

I'll test adding a dummy href.location, that immediately returns a HTTP 200 instead of the HTTP 301, to see if I still see the following log after the sleep().

```
      await sleep(100);
      player.log.debug('After performing the auth redirect and sleep. With persisted logging on in the browser, this log should not show.');
      // This Error should not happen, as the user is redirected to the auth page.
      throw Error('Video not found and user is not authenticated. Try to log in.');
      ```

Reply:
> I'll test adding a dummy location.href, that immediately returns a HTTP 200 instead of the HTTP 301, to see if I still see the following log after the sleep().

In Safari, I verified that it's the HTTP 301 that causes the havoc. The HTTP 200 URL displays the log line before the sleep and not the log line after the sleep. The HTTP 301 URL displays the log line before the sleep, the one after the sleep, the error log, and the flash of red error on the page prior to the redirect.

Test by adding and not adding the redirect param .../watch.html?id=<mpId>&redirect=301

```
      if (!me.roles.includes('ROLE_USER')) {
        player.log.info('Video not found and user is not authenticated. Try to log in.');
+            if( utils.getUrlParameter('redirect') === '301' ) {
+               // Redirect to a URL that returns a HTTP 301
+               location.href ='https://www.iana.org/domains/example';
+               player.log.debug('Redirected to HTTP 301 before sleep');
+               await sleep(100);
+               player.log.debug('Redirected to HTTP 301 after sleep');
+            } else {
+               // Redirect to a URL that returns a HTTP 200
+               window.location.replace('https://example.com');
+               player.log.debug('Redirected to HTTP 200 before sleep');
+               await sleep(100);
+               player.log.debug('Redirected to HTTP 200 after sleep');
+            }
         // This Error should not happen, as the user is redirected to the auth page.
        throw Error('Video not found and user is not authenticated. Try to log in.');
      }
     
```

Therefore, this pull fixes the case when the href is to a page that returns a HTTP 200.

Reply:
Thanks @karendolan for testing it! I updated the PR with a non resolving promise.
Please test it!

Reply:
@karendolan I added a spinner while redirecting. This should solve your problem.

Reply:
@miesgre this works great for me!

I can approve it, but I can't merge it because there is a failing github actions Playright test that isn't related to the code but to the target test video.

```
    Error: Timed out 5000ms waiting for expect(locator).toHaveTitle(expected)

    Locator: locator(':root')
    Expected string: "Dual-Stream Demo - No series | Opencast"
    Received string: "Paella Player 7"
    ```